src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {storage.save(addressBook);}
```

Comment:
Please do not put them all into one line, follow the Java coding standards shown to you before